clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -231,9 +530,26 @@ func getSelectorFromRuntimeObject(obj runtime.Object) (labels.Selector, error) {
 	}
 }
 
+func getSpecFromRuntimeObject(obj runtime.Object) (interface{}, error) {
+	switch typed := obj.(type) {
+	case *corev1.ReplicationController:
+		return typed.Spec, nil
+	case *extensions.ReplicaSet:
+		return typed.Spec, nil
+	case *extensions.Deployment:
+		return typed.Spec, nil
+	case *extensions.DaemonSet:
+		return typed.Spec, nil
+	case *batch.Job:
+		return typed.Spec, nil
+	default:
+		return nil, fmt.Errorf("unsupported kind when getting selector: %v", obj)
+	}
+}
+
```

Comment:
Would it be possible to split this PR a bit so that it's smaller.
Maybe those helper functions and objectChecker struct with its methods?

Reply:
Runtime object related functions moved to #270 .